api/endpoint/lending.go

```diff
@@ -0,0 +1,111 @@
+package endpoint
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/trustwallet/blockatlas/pkg/blockatlas"
+)
+
+// @Summary Get Lending Providers Info.
+// @ID lending_providers
+// @Description Get lending providers, their info, and supported assets.
+// @Produce json
+// @Tags Lending
+// @Success 200 {object} blockatlas.DocsResponse Docs: []blockatlas.LendingProvider
+// @Router /v1/lending/providers [get]
+func HandleLendingProviders(c *gin.Context, apis map[string]blockatlas.LendingAPI) {
+	p, err := getProviders(apis)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+	c.JSON(http.StatusOK, blockatlas.DocsResponse{Docs: &p})
+}
+
+// @Summary Get Asset infos, with yield rates.
+// @ID lending_providers
+// @Description Get lending rates, for one or more assets, of a provider.
+// @Accept json
+// @Produce json
+// @Tags Lending
+// @Param provider path string true "Lending provider name"
+// @Param request body blockatlas.RatesRequest true "Request, containing one or more assets (token symbols)"
+// @Success 200 {object} blockatlas.DocsResponse Docs: []blockatlas.AssetInfo
+// @Router /v1/lending/assets/:provider/:asset [get]
+func HandleLendingAssets(c *gin.Context, apis map[string]blockatlas.LendingAPI) {
+	provider, ok := c.Params.Get("provider")
+	if !ok {
+		c.JSON(http.StatusNotFound, gin.H{"error": "Fatal: missing provider"})
+		return
+	}
+	asset, _ := c.Params.Get("asset")
+	p, err := getAssets(provider, asset, apis)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+	c.JSON(http.StatusOK, blockatlas.DocsResponse{Docs: p})
+}
+
+// @Summary Get Account Contracts.
+// @ID lending_providers
+// @Description Get lending constracts, for one or more adresses, one or more assets (at one provider).
+// @Accept json
+// @Produce json
+// @Tags Lending
+// @Param provider path string true "Lending provider name"
+// @Param request body blockatlas.RatesRequest true "Request, containing one or more assets (token symbols)"
+// @Success 200 {object} blockatlas.DocsResponse Docs: []blockatlas.AccountLendingContracts
+// @Router /v1/lending/account/:provider [post]
+func HandleLendingAccount(c *gin.Context, apis map[string]blockatlas.LendingAPI) {
+	provider, ok := c.Params.Get("provider")
+	if !ok {
+		c.JSON(http.StatusNotFound, gin.H{"error": "Fatal: missing provider"})
+		return
+	}
+	var req blockatlas.AccountRequest
+	err := c.BindJSON(&req)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Invalid request: " + err.Error()})
+		return
+	}
+	p, err := getAccounts(provider, req, apis)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
```

Comment:
what if `getAccounts()` contains sensitive info - it will print that? 

Follow previous example: https://github.com/trustwallet/blockatlas/pull/1124/files#diff-3f5da5f5e56ba0ba382ec8be51046f8cR65

Reply:
Good consideration, haven't thought on that.  OTOH, it is useful to include upstream error, much better than digging for the error in the server logs.
I changed to constant msg.